### ISSUE TITLE
Cannot load wsgi_module

### ISSUE THREAD

**Issue by echaos** [GrahamDumpleton/mod_wsgi#268]
python2.7(64bit) apache(64 bit)
code in the http.conf
`LoadFile "C:\Windows\System32\python27.dll"`
`LoadModule wsgi_module "c:/python27/lib/site-packages/mod_wsgi/server/mod_wsgi.pyd"`
`WSGIPythonHome "c:/python27`

error message:
`Sorry,
There is 'LoadModule wsgi_module modules/mod_wsgi.so' line in httpd.conf file
   but there no 'mod_wsgi.so' file in apachex.y.z/modules/ directory.
Switch cancelled`

**Comment by GrahamDumpleton** [comment]
Take out:

```
LoadModule wsgi_module modules/mod_wsgi.so
```

The ``LoadFile/LoadModule/WSGIPythonHome`` you show is what you want.

**Comment by echaos** [comment]
@GrahamDumpleton It is weird I cannot find the line `LoadModule wsgi_module modules/mod_wsgi.so` in http.conf. By the way, I reinstalled the apache downloaded from ApacheLounge and installed mod_wsgi again. And the error message became 
`C:/Apache24/conf/httpd.conf: Cannot load c:/python27/lib/site-packages/mod_wsgi/server/mod_wsgi.pyd into server: The specified module could not be found`

**Comment by GrahamDumpleton** [comment]
That often means that you are mixing 32 bit and 64 bit versions of software. Ensure that Apache and Python are both 64 bit, and that the compiler available when installing mod_wsgi using ``pip`` was also 64 bit, and not 32 bit. The compiler also needs to be Visual Studio VC9 compiler. Newer versions may not work.

Overall, I would recommend using Python 3.6 and Visual Studio VC14 compiler as they are still current. Python 2.7 causes too many problems because of dependence on really old compiler version.

**Comment by echaos** [comment]
@GrahamDumpleton how to specify a VC9 compiler?

**Comment by GrahamDumpleton** [comment]
You need to have installed the VC9 compiler. It is ancient and why suggest use Python 3.6. You can find details about compilers you must use with different Python versions at:

* https://wiki.python.org/moin/WindowsCompilers

**Comment by GrahamDumpleton** [comment]
Closing as no further followup. Re-open if still an issue and have updated to more recent Python.

**Comment by Ajinkya-Suranis** [comment]
> You need to have installed the VC9 compiler. It is ancient and why suggest use Python 3.6. You can find details about compilers you must use with different Python versions at:
> 
> * https://wiki.python.org/moin/WindowsCompilers

@GrahamDumpleton  I am facing a similar issue on my windows 10 machine:

C:\Apache24\bin>mod_wsgi-express module-config
LoadModule wsgi_module "C:/Python27/lib/site-packages/mod_wsgi/server/mod_wsgiNone"
WSGIPythonHome "C:/Python27"

I inserted the output of above command (after replacing 'None' with '.pyd') in httpd.conf file.
But I'm getting below error:

C:\Apache24\bin>httpd.exe -k start
httpd.exe: Syntax error on line 537 of C:/Apache24/conf/httpd.conf: Cannot load C:/Python27/lib/site-packages/mod_wsgi/server/mod_wsgi.pyd into server: %1 is not a valid Win32 application.

I ensured that all components are 64-bit and python is installed for all users.
Do you think moving to python3.x will fix this error?
Your help is appreciated. Thanks!

**Comment by GrahamDumpleton** [comment]
Definitely move to Python 3.6 or 3.7. It has become near impossible to get Python 2.7 to work any more on Windows.

**Comment by Ajinkya-Suranis** [comment]
> Definitely move to Python 3.6 or 3.7. It has become near impossible to get Python 2.7 to work any more on Windows.

Thanks for quick response! I will try that.

**Comment by tksundar** [comment]
I installed mod_wsgi today and could generate the configs correctly. I have python 3.8. it initially complained about VS c++ . I installed the latest and could install the module. Then I generated the configs and copied the output to httpd.conf as shown below
LoadModule wsgi_module "/users/tksra/pycharmprojects/mysite/venv/lib/site-packages/mod_wsgi/server/mod_wsgi.cp38-win_amd64.pyd"
WSGIPythonHome "/users/tksra/pycharmprojects/mysite/venv"
The paths are good and the .pyd file exists. But apache still gives this error
The Apache service named  reported the following error:
>>> httpd.exe: Syntax error on line 188 of C:/apache/Apache24/conf/httpd.conf: Cannot load c:/users/tksra/pycharmprojects/mysite/venv/lib/site-packages/mod_wsgi/server/mod_wsgi.cp38-win_amd64.pyd into server: The specified module could not be found.  
I am running 64 bit windows and python and VC++
What could be the issue?
